### DISCUSSION TITLE
Should we be worried about the xz backdoor?

### DISCUSSION THREAD

**Discussion by dynnian** [void-linux/void-packages#49614] [General]
As far as I know, Void Linux shouldn't be affected by this even if we are building xz & liblzma from the release tarballs (which we are), because according to Andres Freund [here](https://www.openwall.com/lists/oss-security/2024/03/29/4).

> openssh does not directly use liblzma. However debian and several other
distributions patch openssh to support systemd notification, and libsystemd
does depend on lzma.

Right now Void is shipping version 5.6.0 (which is an affected version) with #49444 to bump it to 5.6.1. Should we downgrade?

**Comment by dynnian** [comment]
The Arch guys said this too [here](https://archlinux.org/news/the-xz-package-has-been-backdoored/):

> Arch does not directly link openssh to liblzma, and thus this attack vector is not possible. ... However, out of an abundance of caution, we advise users to remove the malicious code from their system by upgrading either way. This is because other yet-to-be discovered methods to exploit the backdoor could exist.

**Reply by dynnian** [reply]
For more info here are the links:

RedHat Report: https://www.redhat.com/en/blog/urgent-security-alert-fedora-41-and-rawhide-users
Debian Report: https://lists.debian.org/debian-security-announce/2024/msg00057.html
CVE: https://access.redhat.com/security/cve/CVE-2024-3094

**Comment by classabbyamp** [comment]
> Right now Void is shipping version 5.6.0 (which is an affected version) with https://github.com/void-linux/void-packages/pull/49444 to bump it to 5.6.1.

[no we are not](https://github.com/void-linux/void-packages/blob/master/srcpkgs/xz/template#L4)

void's sshd (or its dependencies) do not link against liblzma, so under the current knowledge, the attack is not possible.

**Reply by dynnian** [reply]
Didn't see that there was a commit to reverse to 5.4.X, yesterday I made an install and I saw 5.6.0, glad to see it was fixed very quickly

**Reply by dynnian** [reply]
I guess if we stay in the 5.4 series we won't have an issue, so I'm closing for now.
